kuma/wiki/views/translate.py

```diff
@@ -214,7 +217,24 @@ def translate(request, document_slug, document_locale, revision_id=None):
                         pass
 
                 rev_form.save(doc)
+                # If this is an Ajax POST, then return a JsonResponse
+                if request.is_ajax():
+                    data = {
+                        'error': False,
+                        'new_revision_id': rev_form.instance.id,
+                    }
+
+                    return JsonResponse(data)
                 return redirect(doc)
+            else:
+                # If this is an Ajax POST, then return a JsonResponse with error
+                if request.is_ajax():
+                    data = {
+                        "error": True,
+                        "error_message": mark_safe(rev_form.errors['current_rev'][0]),
+                        "new_revision_id": rev_form.instance.id,
+                    }
+                    return JsonResponse(data=data, status=500)
```

Comment:
The convention is to use a 200 status code if the JSON is returned without a problem,  even if what the JSON says is `{"error": True}`. Non 200 status codes are reserved for when something goes wrong producing the JSON.